### PR TITLE
Migrate current users to newer version

### PR DESCRIPTION
**Pull request by HarshwardhanSingh**
Adds a rake task to rename question_groups to surveys table.
Adds instructions in readme for upgrading

Let me know if you want different for the rake task

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,6 +28,13 @@ And if you want to customize rapidfire views, you can do
 
     $ bundle exec rails generate rapidfire:views
 
+### Upgrading?
```
Comment (iffyuva) [inline README.md]: we already have a upgrading section i guess

Comment (iffyuva) [inline README.md]: better to mention from which version user is upgrading, and to which version.

File: lib/generators/rapidfire/templates/migrations/rename_answer_groups_to_attempts.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class RenameAnswerGroupsToAttempts < ActiveRecord::Migration[5.0]
```
Comment (iffyuva) [inline lib/generators/rapidfire/templates/migrations/rename_answer_groups_to_attempts.rb]: lets not make these migrations rails 5 only

**Comment by iffyuva** [comment]
overall PR looks good. let me try this out locally once

File: README.md (hunk 1 above)
Comment (HarshwardhanSingh) [inline README.md]: Fixed!!

File: lib/generators/rapidfire/templates/migrations/rename_answer_groups_to_attempts.rb (hunk 2 above)
Comment (HarshwardhanSingh) [inline lib/generators/rapidfire/templates/migrations/rename_answer_groups_to_attempts.rb]: Fixed!!

File: README.md [hunk 3]
```diff
@@ -192,6 +192,15 @@ The typical flow about how to use this gem is:
 
 
 ## Notes on upgrading
+
+##### Upgrading from 2.2.0 to 3.2.0
+
+If you are upgrading you need to rename your `rapidfire_question_groups` to `rapidfire_surveys` and `rapidfire_answer_groups` to `rapidfire_attempts`. Run the given task to do that for you.
+
+```shell
+    $ rake rapidfire:upgrade:migrations:from210to330
```
Comment (iffyuva) [inline README.md]: rake task says from `210 to `330`, and heading says from `2.2.0 to `3.2.0`. the release that we are going to make is `3.0.0` :D

**Comment by iffyuva** [comment]
@HarshwardhanSingh bunch of other things:

- in `rapidfire_answers` table, rename `answer_group_id` to `attempt_id`
- in `rapidfire_attempts` table, rename `question_group_id` to `survey_id`
- in `rapidfire_questions` table, rename `question_group_id` to `survey_id`

please let me know when these changes are done.

**Comment by HarshwardhanSingh** [comment]
@yuva I have already renamed the above mentioned columns in migrations tables.

File: README.md (hunk 3 above)
Comment (HarshwardhanSingh) [inline README.md]: should it be 2.0.0 to 3.0.0?

**Comment by iffyuva** [comment]
@HarshwardhanSingh where? link?

**Comment by HarshwardhanSingh** [comment]
[rapidfire_answers](https://github.com/code-mancers/rapidfire/blob/migrate-current-users-to-newer-version/db/migrate/20130502195504_create_rapidfire_answers.rb)
[rapidfire_attempts](https://github.com/code-mancers/rapidfire/blob/migrate-current-users-to-newer-version/db/migrate/20130502195415_create_rapidfire_answer_groups.rb)
[rapidfire_questions](https://github.com/code-mancers/rapidfire/blob/migrate-current-users-to-newer-version/db/migrate/20130502195310_create_rapidfire_questions.rb)

**Comment by iffyuva** [comment]
@HarshwardhanSingh im talking about existing users, and not new users. i need migrations for existing users to rename those 3 columns.

**Comment by HarshwardhanSingh** [comment]
@iffyuva okay, got it

File: README.md (hunk 3 above)
Comment (HarshwardhanSingh) [inline README.md]: @iffyuva

**Pull request merged**